File: archive/Source/2019/Day1_2.exs
fuelRequired = fn 
    (x, _) when x < 9 -> 0
    (x, recurse) -> (div x, 3) - 2 + recurse.((div x, 3) - 2, recurse)
end

IO.puts(
File.stream!("Day1.txt", [], :line)
    |> Enum.map(&String.trim(&1))
    |> Enum.filter(&(&1 != ""))
    |> Enum.map(&fuelRequired.(String.to_integer(&1), fuelRequired))
    |> Enum.sum
)

File: archive/Source/2021/Day1_2.py
file = open("Day1.txt", "r")

increase_count = 0
data = []

for line in file:
  data.append(int(line))

last = 100000
for i in range(len(data)-2):
  sum = 0
  for j in range(3):
    sum += data[i+j]

  if sum > last: increase_count += 1
  last = sum

print (increase_count)
